### PR TITLE
Add arch test for preview API usage

### PR DESCRIPTION
**Pull request by marcoferrer**
# Description 
This PR introduces the concept of architecture tests to the project. The first test implemented verifies that any method or class that is annotated `@Preview` is also annotated with `@Deprecated`.

The library used to implement the tests is [ArchUnit](https://github.com/TNG/ArchUnit)

The following is an example of the output created when this test fails.
image

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn clean compile` locally. This may reformat your code, commit those changes.
- [x] Run `mvn -D enable-ci clean install site` locally. If this command doesn't succeed, your change will not pass CI.

# When creating a PR: 

- [x] Fill in the "Description" above. 
- [x] Enable "Allow edits from maintainers".

### CODE REVIEW

**Comment by marcoferrer** [comment]
@bitwiseman Based on the discussion we had in this PR https://github.com/hub4j/github-api/pull/960

I decided to implement some architecture tests to cover the `@Deprecated` requirement via CI.

Thoughts?

File: src/test/java/org/kohsuke/github/ArchTests.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+import org.junit.Test;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+
+public class ArchTests {
+
+    private final JavaClasses classFiles = new ClassFileImporter()
```
Comment (sullis) [inline src/test/java/org/kohsuke/github/ArchTests.java:19]: "classFiles" could be a static variable.

Then I would add this:

``
    @BeforeClass
    public static void beforeClass() {
        assertTrue(classFiles.size() > 0);
    }
```

**Review by sullis — CHANGES_REQUESTED** [review]

File: src/test/java/org/kohsuke/github/ArchTests.java (hunk 1 above)
Comment (bitwiseman) [reply src/test/java/org/kohsuke/github/ArchTests.java:19]: @sullis 
Which way do you think it better?

**Review by bitwiseman — APPROVED** [review]
I like this. Seems reasonable.  Please address the one comment so far and then I'll merge.

**Comment by marcoferrer** [comment]
@bitwiseman addressed feedback, simplified implementation, added another rule to check for preview APIs in enum fields

**Pull request merged**